### PR TITLE
refactor: registry cache

### PR DESCRIPTION
**Pull request by mouuii**

### CODE REVIEW

File: pkg/cluster/registry/factory/factory.go [hunk 1]
```diff
@@ -10,36 +10,35 @@ import (
...
-	return &factory{
-		registryCache: registryCache,
+func NewRegistryCache() *RegistryCache {
+	return &RegistryCache{
+		&sync.Map{},
 	}
 }
 
-func (f *factory) GetRegistryByConfig(ctx context.Context, config *registry.Config) (registry.Registry, error) {
+func (f *RegistryCache) GetRegistryByConfig(ctx context.Context, config *registry.Config) (registry.Registry, error) {
```
Comment (kilosonc) [inline pkg/cluster/registry/factory/factory.go:33]: Is uppercase needed?

Comment (mouuii) [reply pkg/cluster/registry/factory/factory.go]: done

File: pkg/cluster/registry/factory/factory.go [hunk 2]
```diff
@@ -10,36 +10,35 @@ import (
 
 var (
 	// Fty is the global registry factory
-	Fty = NewFactory()
+	Fty = NewRegistryCache()
 )
 
 // nolint
 //
 //go:generate mockgen -source=$GOFILE -destination=../../../../mock/pkg/cluster/registry/factory/factory_mock.go -package=mock_factory
```
Comment (xuzhu-591) [inline pkg/cluster/registry/factory/factory.go:18]: Could you run "go generate ./..." to re-generate mock files?

File: pkg/cluster/registry/factory/factory.go [hunk 3]
```diff
@@ -10,36 +10,35 @@ import (
...
-	return &factory{
-		registryCache: registryCache,
+func NewRegistryCache() *registryCache {
+	return &registryCache{
+		&sync.Map{},
 	}
 }
 
-func (f *factory) GetRegistryByConfig(ctx context.Context, config *registry.Config) (registry.Registry, error) {
+func (f *registryCache) GetRegistryByConfig(ctx context.Context, config *registry.Config) (registry.Registry, error) {
```
Comment (xuzhu-591) [inline pkg/cluster/registry/factory/factory.go:33]: f => r ?

Comment (mouuii) [reply pkg/cluster/registry/factory/factory.go]: fixed

File: pkg/cluster/registry/factory/factory.go (hunk 2 above)
Comment (mouuii) [reply pkg/cluster/registry/factory/factory.go:18]: ok

**Review by xuzhu-591 — APPROVED** [review]

**Pull request merged**